### PR TITLE
Second attempt to fix MatrixElement Printer #11839

### PR DESCRIPTION
**Pull request by Alsheh**
In this pull request, I rewrote [the older pull request which fixes issue #11839](https://github.com/sympy/sympy/pull/11839). This solution distributes the element access over the expression.

@asmeurer and @Upabjojr  please review.

### CODE REVIEW

**Comment by Alsheh** [comment]
Any ideas why the code is failing this test case?

File: sympy/matrices/expressions/matexpr.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from sympy.functions import conjugate, adjoint
 from sympy.matrices import ShapeError
 from sympy.simplify import simplify
+from sympy.printing.precedence import precedence, PRECEDENCE
```
Comment (Upabjojr) [inline sympy/matrices/expressions/matexpr.py]: This line should not be here.

File: sympy/matrices/expressions/matexpr.py [hunk 2]
```diff
@@ -372,6 +373,41 @@ def _eval_derivative(self, v):
         from sympy import KroneckerDelta
         return KroneckerDelta(self.args[1], v.args[1])*KroneckerDelta(self.args[2], v.args[2])
 
+    def _MatrixElement_expansion(self, printer, elementAccess):
```
Comment (Upabjojr) [inline sympy/matrices/expressions/matexpr.py]: No printers in this module.

File: sympy/matrices/expressions/matexpr.py [hunk 3]
```diff
@@ -372,6 +373,41 @@ def _eval_derivative(self, v):
...
+        # if the MatrixElement object is an instance of MatrixSymbol,
+        # simply, return the concatenation of the matrix element and elementAccess
+        if isinstance(self.parent, MatrixSymbol):
+             return printer._print(self.parent) + elementAccess
+
+        # get the terms from the MatrixElement object
+        terms = list(self.parent.args)
+        # creat an empty list in which the intermediate result is stored
+        l = []
+        # loop over the terms and concatenate the term and elementAccess
```
Comment (Upabjojr) [inline sympy/matrices/expressions/matexpr.py]: This is way too complicated, this method should just rewrite `MatrixElement(MatAdd(a, b, c,  ...), ...)` to `Add(MatrixElement(a, ...), MatrixElement(b, ...), ... )`

File: sympy/matrices/expressions/matexpr.py [hunk 4]
```diff
@@ -372,6 +373,41 @@ def _eval_derivative(self, v):
...
+                sign = "+"
+            # concatenate the term and elementAccess and add them to l
+            l.extend([sign, t + elementAccess])
+
+        # if the first term is positive, remove its sign from l
+        sign = l.pop(0)
+        if sign == '+':
+            sign = ""
+
+        return sign + ' '.join(l)
```
Comment (Upabjojr) [inline sympy/matrices/expressions/matexpr.py]: Return should be an `Add` or `MatrixElement`.

**Comment by Upabjojr** [comment]
> Any ideas why the code is failing this test case #20236.20?

Probably a bug in Travis, just ignore it.

**Comment by jksuom** [comment]
That is a known issue: https://github.com/sympy/sympy/issues/11684

**Comment by madan96** [comment]
@jksuom I think restarting the build for test case #20236.20, the test case might pass. It did in my case.

**Comment by czgdp1807** [comment]
@Alsheh Thanks for the contribution. Would you like to continue this?

**Comment by Alsheh** [comment]
@czgdp1807 thanks for the follow up! I almost lost context of this PR and will most likely not be updated. Closing this PR for now.

**Pull request closed without merging**